### PR TITLE
Lrn 39610/feature/add signature v2

### PR DESCRIPTION
**Pull request by markus-liang-lrn**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,6 @@ target
 Dist
 .idea
 learnositysdk.iml
+.DS_Store
+*/.DS_Store
+docs/quickstart/assessment/webapps/
```
Comment (bhavya-shukla-lrn) [inline .gitignore:7]: Why is this added to git ignore "docs/quickstart/assessment/webapps/"
add a new line in the end.

File: docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        <script src="https://items.learnosity.com/?latest"></script>
+
+        <!-- Initiate Items API assessment rendering, using the JSON blob of signed params. -->
+        <script>
+            var itemsApp = LearnosityItems.init(
+                <%= app.initOptions(request.getServerName()) %>
+            );
+        </script>
+    </body>
+</html>
```
Comment (bhavya-shukla-lrn) [inline docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp:33]: add a new line

File: docs/quickstart/assessment/src/main/webapp/ReportsApi.jsp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        <script src="https://reports.learnosity.com/?latest"></script>
+
+        <!-- Initiate Questions API assessment rendering, using the JSON blob of signed params. -->
+        <script>
+            var reportsApp = LearnosityReports.init(
+                <%= app.initOptions(request.getServerName()) %>
+            );
+        </script>
+    </body>
+</html>
```
Comment (bhavya-shukla-lrn) [inline docs/quickstart/assessment/src/main/webapp/ReportsApi.jsp:33]: add new line

File: docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/AuthorApp.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            Init init = new Init("author", security, secret, request);
+            return init.generate();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "";
+        }
+    }
+
+    Map<String, String> createSecurityObject(String domain) {
+        var secret = new HashMap();
```
Comment (bhavya-shukla-lrn) [inline docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/AuthorApp.java:25]: should this be security instead of secret?

File: docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/ItemsApp.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            Init init = new Init("items", security, secret, request);
+            return init.generate();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "";
+        }
+    }
+
+    Map<String, String> createSecurityObject(String domain) {
+        var secret = new HashMap();
```
Comment (bhavya-shukla-lrn) [inline docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/ItemsApp.java:24]: Same as above ? security??

File: .gitignore (hunk 1 above)
Comment (markus-liang-lrn) [reply .gitignore]: that is for compiled folder

File: docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/QuestionsApp.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            Init init = new Init("questions", security, secret, request);
+            return init.generate();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "";
+        }
+    }
+
+    Map<String, String> createSecurityObject(String domain) {
+        var secret = new HashMap();
```
[the remark this answers is not in the document]
Comment (markus-liang-lrn) [reply docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/QuestionsApp.java]: done

File: docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp (hunk 2 above)
Comment (markus-liang-lrn) [reply docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp]: done

File: docs/quickstart/assessment/src/main/webapp/QuestionsApi.jsp [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        <script src="https://questions.learnosity.com/?latest"></script>
+
+        <!-- Initiate Questions API assessment rendering, using the JSON blob of signed params. -->
+        <script>
+            var questionsApp = LearnosityApp.init(
+                <%= app.initOptions(request.getServerName()) %>
+            );
+        </script>
+    </body>
+</html>
```
[the remark this answers is not in the document]
Comment (markus-liang-lrn) [reply docs/quickstart/assessment/src/main/webapp/QuestionsApi.jsp]: done

File: docs/quickstart/assessment/src/main/webapp/ReportsApi.jsp (hunk 3 above)
Comment (markus-liang-lrn) [reply docs/quickstart/assessment/src/main/webapp/ReportsApi.jsp]: done

File: docs/quickstart/assessment/src/main/webapp/index.jsp [hunk 8]
```diff
@@ -1,33 +1,16 @@
...
-            );
-        </script>
+        <ul>
+            <li><a href="ItemsApi.jsp">Items API</a></li>
+            <li><a href="QuestionsApi.jsp">Questions API</a></li>
+            <li><a href="AuthorApi.jsp">Author API</a></li>
+            <li><a href="ReportsApi.jsp">Reports API</a></li>
+        </ul>
     </body>
 </html>
```
[the remark this answers is not in the document]
Comment (markus-liang-lrn) [reply docs/quickstart/assessment/src/main/webapp/index.jsp]: done

File: docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/AuthorApp.java (hunk 4 above)
Comment (markus-liang-lrn) [reply docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/AuthorApp.java]: done

File: docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/ItemsApp.java (hunk 5 above)
Comment (markus-liang-lrn) [reply docs/quickstart/assessment/src/main/java/com/learnosity/quickstart/ItemsApp.java]: done

File: docs/quickstart/assessment/src/main/webapp/AuthorApi.jsp [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        <script src="https://authorapi.learnosity.com/?latest"></script>
+
+        <!-- Initiate Author API assessment rendering, using the JSON blob of signed params. -->
+        <script>
+            var authorApp = LearnosityAuthor.init(
+                <%= app.initOptions(request.getServerName()) %>
+            );
+        </script>
+    </body>
+</html>
```
[the remark this answers is not in the document]
Comment (markus-liang-lrn) [reply docs/quickstart/assessment/src/main/webapp/AuthorApi.jsp]: done

**Review by bhavya-shukla-lrn — APPROVED** [review]

File: docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+<html>
+    <head><link rel="stylesheet" type="text/css" href="css/style.css"></head>
+    <body>
+        <h1>Standalone Assessment Example - Items API</h1>
+
+        <!-- Items API will render the assessment app into this div. -->
+        <div id="learnosity_assess"></div>
+
+        <!-- Load the Items API library. -->
+        <script src="https://items.learnosity.com/?latest"></script>
```
Comment (TheRealEdDawson) [inline docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp:24]: @markus-liang-lrn In here, let's change this "latest" value to "latest-lts", because I really want to emphasise that clients should be selecting an LTS version.

File: docs/quickstart/assessment/src/main/webapp/QuestionsApi.jsp [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+<html>
+    <head><link rel="stylesheet" type="text/css" href="css/style.css"></head>
+    <body>
+        <h1>Standalone Assessment Example - Question API</h1>
+
+        <!-- Questions API will render the Question app into this span. -->
+        <span class="learnosity-response question-60001"></span>
+
+        <!-- Load the Questions API library. -->
+        <script src="https://questions.learnosity.com/?latest"></script>
```
Comment (TheRealEdDawson) [inline docs/quickstart/assessment/src/main/webapp/QuestionsApi.jsp:24]: @markus-liang-lrn In here, let's change this "latest" value to "latest-lts", because I really want to emphasise that clients should be selecting an LTS version.

File: docs/quickstart/assessment/src/main/webapp/ReportsApi.jsp [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+<html>
+    <head><link rel="stylesheet" type="text/css" href="css/style.css"></head>
+    <body>
+        <h1>Standalone Assessment Example - Reports API</h1>
+
+        <!-- Reports API will render the author app into this span. -->
+        <span class="learnosity-report" id="session-detail"></span>
+
+        <!-- Load the Reports API library. -->
+        <script src="https://reports.learnosity.com/?latest"></script>
```
Comment (TheRealEdDawson) [inline docs/quickstart/assessment/src/main/webapp/ReportsApi.jsp:24]: @markus-liang-lrn In here, let's change this "latest" value to "latest-lts", because I really want to emphasise that clients should be selecting an LTS version.

File: docs/quickstart/assessment/src/main/webapp/AuthorApi.jsp [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+<html>
+    <head><link rel="stylesheet" type="text/css" href="css/style.css"></head>
+    <body>
+        <h1>Standalone Assessment Example - Author API</h1>
+
+        <!-- Items API will render the assessment app into this div. -->
+        <div id="learnosity-author"></div>
+
+        <!-- Load the Author API library. -->
+        <script src="https://authorapi.learnosity.com/?latest"></script>
```
Comment (TheRealEdDawson) [inline docs/quickstart/assessment/src/main/webapp/AuthorApi.jsp:24]: @markus-liang-lrn In here, let's change this "latest" value to "latest-lts", because I really want to emphasise that clients should be selecting an LTS version.

**Review by TheRealEdDawson — APPROVED** [review]
I left a few minor comments, but generally, looks great, guys. Please go ahead!

File: docs/quickstart/assessment/src/main/webapp/AuthorApi.jsp (hunk 13 above)
Comment (bhavya-shukla-lrn) [reply docs/quickstart/assessment/src/main/webapp/AuthorApi.jsp:24]: @TheRealEdDawson I guess , this will apply to all sdks ?

File: docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp (hunk 10 above)
Comment (bhavya-shukla-lrn) [reply docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp:24]: @TheRealEdDawson I guess , this will apply to all sdks ?

File: docs/quickstart/assessment/src/main/webapp/QuestionsApi.jsp (hunk 11 above)
Comment (bhavya-shukla-lrn) [reply docs/quickstart/assessment/src/main/webapp/QuestionsApi.jsp:24]: @TheRealEdDawson I guess , this will apply to all sdks ?

File: docs/quickstart/assessment/src/main/webapp/ReportsApi.jsp (hunk 12 above)
Comment (bhavya-shukla-lrn) [reply docs/quickstart/assessment/src/main/webapp/ReportsApi.jsp:24]: @TheRealEdDawson I guess , this will apply to all sdks ?

**Review by TheRealEdDawson — APPROVED** [review]
Looks great, thanks @bhavya-shukla-lrn . Please go ahead with it.

File: docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp (hunk 10 above)
Comment (TheRealEdDawson) [reply docs/quickstart/assessment/src/main/webapp/ItemsApi.jsp]: Yes, I think so.

**Review by TheRealEdDawson — APPROVED** [review]
Looks good @markus-liang-lrn @bhavya-shukla-lrn @markkellyeire , I made some minor edits to text in README.md.

File: CHANGELOG.md [hunk 14]
```diff
@@ -6,6 +6,10 @@
 
 ## [Unreleased]
 
+## [v0.16.4] - 2023-06-23
```
[the remark this answers is not in the document]
Comment (TheRealEdDawson) [reply CHANGELOG.md:9]: @markus-liang-lrn @bhavya-shukla-lrn @markkellyeire I think we can probably disable this Codacy rule ("Lists should be surrounded by blank lines"), since it goes against our own convention in this file?
